### PR TITLE
Make cmd palette 'go back' button return to previously selected action

### LINKED ISSUE #13457: Command palette  goes to the top when you go back (<) from a submenu
As the subject says, if I open a submenu of the command palette (e.g., "New tab") and go back via the <-button, I'm back at the top of the command palette.  I'd prefer to be back where I was.  It's not a bug, but the bahavior could be made more friendly.

### PR DESCRIPTION
**Pull request by JerBast**
The behaviour of the 'go back' button in the command palette was changed to return to the previously selected element rather than the root.

Instead of returning to the root, the go back button now returns to the previously selected item in the filtered action list. The previously selected item is selected by default and the view is scoped to the item.

## Validation Steps Performed
Manually tested by going back and forth between nested actions in the command palette.

Closes #13457

### CODE REVIEW

**Review by DHowett — APPROVED** [review]
I love it, thank you!

It feels pretty economical to search for the last active entry in the list of entries, and to rebuild the nested list like this. It would be good follow-up work to deduplicate the code that generates `_currentNestedCommands` from a `Command`, but I'm not going to ask you to do that now... unless you want bonus points[1]

[1] points don't matter, but i like to pretend they do

**Comment by JerBast** [comment]
Thank you for the review.

I agree that it would be better to deduplicate that fragment. I'll make sure to change it!

File: src/cascadia/TerminalApp/CommandPalette.cpp [hunk 1]
```diff
@@ -536,12 +536,32 @@ namespace winrt::TerminalApp::implementation
...
+            _currentNestedCommands.Clear();
+        }
         _updateFilteredActions();
-        _filteredActionsView().SelectedIndex(0);
+
+        const auto lastSelectedIt = std::find_if(begin(_filteredActions), end(_filteredActions), [&](const auto& filteredCommand) {
+            return filteredCommand.Item().Name() == previousAction.Item().Name();
+        });
+        const auto lastSelectedIndex = static_cast<int32_t>(std::distance(begin(_filteredActions), lastSelectedIt));
+        _scrollToIndex(lastSelectedIt != end(_filteredActions) ? lastSelectedIndex : 0);
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/CommandPalette.cpp:564]: this is clever, I love it

**Review by zadjii-msft — APPROVED** [review]
This is great, thanks for whipping this up!

**Pull request merged**